model/version.go

```diff
@@ -155,6 +156,36 @@ func (self *Version) SetNotActivated() error {
 	)
 }
 
+func (self *Version) SetAborted() error {
```

Comment:
instead of having two very similar functions, we could instead pass in an aborted boolean and set it to that